File: java/src/main/java/com/android/frame/mvc/IBaseListView.java
package com.android.frame.mvc;

import java.util.List;

/**
 * Created by xuzhb on 2021/8/13
 */
public interface IBaseListView<T> extends IBaseView {

    /**
     * 获取加载的起始页码
     */
    int getFirstPage();

    /**
     * 是否是首次加载
     */
    boolean isFirstLoad();

    /**
     * 获取一页加载的个数
     */
    int getLoadSize();

    /**
     * 显示数据
     */
    void showData(int page, List<T> list);

    /**
     * 处理数据
     */
    List<T> convertData(List<T> response);

}

File: kotlin/src/main/java/com/android/frame/mvc/IBaseListView.kt
package com.android.frame.mvc

/**
 * Created by xuzhb on 2021/8/13
 */
interface IBaseListView<T> : IBaseView {

    /**
     * 获取加载的起始页码
     */
    fun getFirstPage(): Int

    /**
     * 是否是首次加载
     */
    fun isFirstLoad(): Boolean

    /**
     * 获取一页加载的个数
     */
    fun getLoadSize(): Int

    /**
     * 显示数据
     */
    fun showData(page: Int, list: MutableList<T>?)

    /**
     * 处理数据
     */
    fun convertData(response: MutableList<T>?): MutableList<T>?

}